### DISCUSSION TITLE
Zrušení MX pro @nette.org na Google

### DISCUSSION THREAD

**Discussion by milo** [nette/nette#1556] [General]
Navrhuji změnit MX DNS záznam pro doménu nette.org na:
```
nette.org MX vps.nette.org
```
Následně změnit TXT SPF záznam:
```
 z: nette.org TXT v=spf1 include:_spf.google.com ip4:78.128.211.35 ip6:2001:718:1:1f:50:56ff:feee:35 ~all
na: nette.org TXT v=spf1 ip4:78.128.211.35 ip6:2001:718:1:1f:50:56ff:feee:35 include:spf.mandrillapp.com -all
```
A vytvořit TXT záznam:
```
mandrill._domainkey.example.com TXT v=DKIM1; k=rsa; p=MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQCrLHiExVd55zd/IQ/J/mRwSRMAocV/hMB3jXwaHH36d9NaVynQFYV8NaWi69c1veUtRzGt7yAioXqLj7Z4TeEUoOLgrKsn8YnckGs9i3B3tVFB+Ch/4mPhXWiNfNdynHWBcPcbJ8kjEQ2U8y78dHZj1YeRXXVvWob2OaKynO8/lQIDAQAB;
```

SPF include a DKIM pro MailChimp vycházím z https://mailchimp.com/developer/transactional/docs/authentication-delivery/

A nakonec na VPS nastavím redirect adres na:
```
david@nette.org => (zruším)
vps@nette.org => (zruším)
honza@nette.org => honza@kreativnilaborator.cz
felix@nette.org => sulcmil@gmail.com
```
Ostatní systémové e-maily, jako je www-data@vps.nette.org a podobné zůstanou směrované na mě.

Done.

**Comment by f3l1x** [comment]
Doplnil jsem tam svuj gmail ucet pro presmerovani. Dekuji.

**Comment by milo** [comment]
@chemix Ty sis nějak v MailChimp verifikoval doménu nette.org?

**Reply by chemix** [reply]
@milo verifikoval jsem email adresu honza@nette.org ... kam dorazil kod a ten jsem zadal do MCh ...

**Reply by milo** [reply]
Aha. To je pouze verifikace adresy, ze které chceš odesílat. A nemáš nějakou zpětnou vazbu, že lidem padají newslettery do spam složky?

**Reply by dg** [reply]
Já ho párkrát ve spamu našel. Ale nemusí to s tím souviset.

**Reply by milo** [reply]
No, vlastně může. Záleží na přísnosti spam filtru. Servery MailChimpu nejsou v SPF pro `nette.org`, tedy se na ně aplikuje `~all` což je soft fail.

**Comment by milo** [comment]
@dg Připravil jsem konfiguraci. Změnil bys prosím MX záznam na `nette.org MX 10 vps.nette.org`? SPF a TXT pro DKIM můžeme vyřešit později. Ty záznamy co jsem psal v úvodu jsou pouze pokud se e-maily rozesílají přes nějaké jejich API.

**Reply by milo** [reply]
Tak v MailChimpu to vyřešili delegací DKIM záznamů na sebe a jejich záznam v SPF tak není potřeba. Stačí Davide, když vytvoříš dva CNAME záznamy:
```
k2._domainkey CNAME 1800 dkim2.mcsv.net
k3._domainkey CNAME 1800 dkim3.mcsv.net
```
Ale ten `k2._domainkey` už existuje jako `A       78.128.211.35`, tedy vps.nette.org, ale nevím proč.

**Reply by dg** [reply]
Jsou tam.

**Reply by milo** [reply]
MX vidím. Starý měl TTL 12 hodin, tak to ověřím až ráno. Ale ty CNAME jsi nadefinoval jako `k2._domainkey.nette.org.nette.org.`

**Reply by milo** [reply]
Anebo ne? Něco mi nesedí... Že ty tam máš někde wildcard `*.nette.org`? :)

**Reply by milo** [reply]
Tak ne, ty CNAME jsou také dobře, sorry. Ale zas musím počkat až vyprší negativní TTL.

**Comment by dg** [comment]
Ještě mě napadá, že se z noreply@nette.org rozesílá pošta z fóra, jen pro úplnost.

**Reply by milo** [reply]
Adresa odesílatele může být cokoliv@nette.org a odejde to v pořádku, jde jen o doručování. Ale odpověď na noreply@nette.org se už nepovede.

**Reply by milo** [reply]
Kdybyste chtěli přeposílat poštu poslanou na cokoliv@nette.org, dejte vědět. Je to 1 řádek v konfiguráku.

**Reply by dg** [reply]
Já mám 12000 nepřečtených emailů, řeším jiný problém :-)

**Comment by milo** [comment]
@f3l1x @chemix Kluci, poslal jsem vám testovací zprávu přes felix@nette.org a honza@nette.org, která už protekla přes VPS. Dorazila?

**Reply by f3l1x** [reply]
Zatim nic.

**Reply by milo** [reply]
Spam složka?

**Reply by f3l1x** [reply]
Jo, uz.

**Reply by milo** [reply]
Prima. Základ funguje. Zapnul jsem ještě [SRS](https://en.wikipedia.org/wiki/Sender_Rewriting_Scheme) a poslal druhý test. Pokud uvidíte někdy jako odesílatele něco jako `From: SRS0=xxxx=yy=gmail.com=miloslav.hula@nette.org` je to v pořádku, pouze váč mail klient nepodporuje dekódování SRS formátu.

**Comment by milo** [comment]
@chemix V MailChimp si můžeš validovat doménu nette.org (Websites -> Domains -> Email Domains) Při validaci vyber jakéhokoliv providera, třeba wordpress.com. Ukazují podle toho pouze návod na přidání DNS záznamů. Po validaci by měly newslettery přestat padat do koše, protože už budou mít validní DKIM podpis.

**Reply by milo** [reply]
@chemix Zkoušel jsi zvalidovat tu doménu?

**Reply by milo** [reply]
Honzo, zkoušel jsi to? Chtěl bych pokročit dál.

**Reply by chemix** [reply]
![obrazek](https://user-images.githubusercontent.com/42802/151729907-d16ad488-8e9f-4504-8ee5-7584f43855b2.png)

asi hotovo

**Reply by milo** [reply]
👍 Prima. Teď by měly být e-maily z Mailchimpu podepsány DKIM podpisem za nette.org doménu a neměly by padat do spamu.

**Comment by dg** [comment]
Pro mě je překvapení, že se takto dají přeposílat emaily. Takže teoreticky, kdybych zrušil Google na grudl.com, mohl bych přeposílat emaily na davidek@gmail.com?

**Reply by milo** [reply]
Jasně. Ale musíš mít nějaký server, na který bude směřovat MX pro grudl.com a který ten forwarding bude dělat.

Nikdy jsem to nevyužil, ale při registraci domény občas takový forwarding nabízí na jejich serverech, takže ani svůj server nepotřebuješ.

**Comment by milo** [comment]
@dg Teď už můžeš upravit TXT záznam se SPF na:
```
nette.org TXT v=spf1 ip4:78.128.211.35 -all
```

IPv6 na serveru nepoužíváme, MailChimp SPF nepotřebuje, takže může být takhle striktní.

**Reply by chemix** [reply]
a za +-14 dni uvidime jak to dopadne s newsletterem :)
